cpp/CMakeLists.txt

```diff
@@ -521,6 +545,10 @@ function(add_library_module NAMESPACE MODULE MODULE_SOURCE_DIRS)
     target_compile_definitions("${NAMESPACE}_${MODULE}" PUBLIC "SPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_${LOGGING_LEVEL}")
     add_dependencies("${NAMESPACE}_${MODULE}" stringify_run)
     target_link_libraries("${NAMESPACE}_${MODULE}" arrow arrow_cuda nvrtc ${CUDART_LIBRARY} cuda ${ZLIB_LIBRARIES} ${Boost_LIBRARIES})
+    if(CUFILE_INSTALLED)
+        target_link_libraries("${NAMESPACE}_${MODULE}" cufile)
+    endif(CUFILE_INSTALLED)
```

Comment:
I don't think we want to do this. For example, we'll likely always build in an environment that has cuFile in it, but a user won't necessarily have it in their runtime environment. We want to try to dynamically load cuFile at runtime via `dlopen`.

Reply:
For some reason I completely forgot why we want to use `dlopen` :facepalm: Thanks for the reminder.
Is there a path that we can use when loading the lib with `dlopen`? I think an environment variable can be useful here, but can't think of anything beyond that.

Reply:
I think we can just rely on the linker search path: https://linux.die.net/man/3/dlopen

If cuFile is placed in a location similar to CUDA libraries that aren't on the default linker search path, I think it's up to the cuFile installation procedure to add it to `LD_LIBRARY_PATH`.